app/assets/javascripts/angular/controllers/dashboard_ctrl.js

```diff
@@ -30,6 +30,31 @@ angular.module("Prometheus.controllers")
     $scope.$broadcast('redrawGraphs');
   }
 
+  $scope.generateStaticDashboard = function(event) {
+    // TODO: When the input with the static dashboard link is visible,
+    // should clicking the "generate" button close the input, or
+    // generate a NEW static link?
+    $scope.generatingPermalink = true;
+    $http.post(window.location.origin + "/static", {
+      'dashboard': {
+        'name': dashboardName,
+        'dashboard_json': {
+          'globalConfig': $scope.globalConfig,
+          'widgets': $scope.widgets
+        }
+      },
+    }).then(function(payload) {
+      $scope.showPermalink = true;
+      $scope.permalink = window.location.origin + payload.data.url;
+      var input = $("[ng-model=permalink]")[0];
+      InputHighlighter(input);
+    }, function() {
+      console.log("failed")
```

Comment:
console.log or alert? i'm thinking alert, as that's how we inform the user a save of the dashboard failed